src/oauth2/bookshare/urls.py

```diff
@@ -0,0 +1,15 @@
+from django.urls import include, path
+from allauth.socialaccount.providers.oauth2.urls import default_urlpatterns
+from pages.views import ReaderIndexView
+#from allauth.socialaccount import views as socialaccount_views
+#import allauth
+
+from .provider import BookshareProvider
+
+urlpatterns = default_urlpatterns(BookshareProvider) + [
+    # Map 'socialaccount_connections' to Clusive's reader page.  This is because
+    # allauth hard codes a reverse('socialaccount_connections') to handle the
+    # end of the workflow triggered by '.../login?process=connect'.  By adding
+    # this path, the end of that workflow will end up on Clusive's reader page.
+    path('reader', ReaderIndexView.as_view(), name='socialaccount_connections')
```

Comment:
This should be a redirect view rather than an extra URL for an existing page.

Reply:
I looked into whether `allauth` supports a "next" request parameter.  It does, using the value as a redirect at the end of the authorization/access-token workflow.  Thus, you can start the workflow with, e.g., `/accounts/bookshare/login?process=connect&next=/dashboard`, and the user will end up on the Dashboard page after successfully connecting to their Bookshare account.  That's actually quite useful: Depending on which page the user initiates authorization, the url can specify an appropriate end point.

However, given the `process=connect` parameter, `allauth` insists on running `reverse('socialaccount_connections')` as part of its connections workflow.  Its `DefaultSocialAccountAdapter` class, specifically its `get_connect_redirect_url()` method, is where happens.  I can find no way to change that behaviour by a simple change of setting or view, other that to map `socialaccount_connections` to a known Clusive url.

The `allauth` [documentation](https://django-allauth.readthedocs.io/en/latest/advanced.html#custom-redirects) says to:
1. subclass `DefaultSocialAccountAdapter` and override its `get_connect_redirect_url()' method
2. set `SOCIALACCOUNT_ADAPTER` to the new class in the project's `settings.py`

Note that 2 is "global" in the sense that the subclass is used for all `allauth` workflows.  In Clusive's case, that means it's also used for Google SSO.  Practically, this is not a problem since Google SSO dos not involve `process=connect`.  But, it is something to keep in mind for other authorization scenarios we might implement in the future.

Another route is to use `allauth`'s default setup for the `socialaccount_connections` endpoint, but that entails incorporating its associated social connections html template, and everything that it imports.  I haven't explored this solution much since I assumed Clusive would not want that `allauth` page to be available to users.

The latest commit uses the subclass/override technique.

Reply:
I had another thought:  Instead of implementing the sublcass, etc., change this particular url pattern to:

> `path('bookshare_connected', views.bookshare_connected, name='socialaccount_connections')`,

where `bookshare_connected()` has a default redirect to somewhere suitable.  But, the presence of a `next=/...`parameter in the original request would have precedence.

Maybe that's what you meant in the first place?